pytorch_lightning/utilities/fetching.py

```diff
@@ -35,34 +38,60 @@
 class AbstractDataFetcher(ABC):
 
     """
-    This class is used to control batch fetching flow.
+    This based class should be used to implement a fault tolerant ``DataFetcher``.
+    It is required to override the ``fetching_function`` with fetching logic.
+    Example::
+        class SimpleDataFetcher(AbstractDataFetcher):
```

Comment:
Need the blank lines for docs to render properly
```suggestion
    It is required to override the ``fetching_function`` with fetching logic.
    
    Example::
    
        class SimpleDataFetcher(AbstractDataFetcher):
```

Reply:
Done !